docs/concepts/linq.rst

```diff
@@ -0,0 +1,268 @@
+LINQ
+====
+
+What is it?
+-----------
+
+LINQ provides language-level querying capabilities and `higher-order function <https://en.wikipedia.org/wiki/Higher-order_function>`_ API to C# and VB (note that F# already has this capability by nature of being a functional programming language).  It is used either via native query syntax,
+
+.. code-block:: c#
+
+	var dotNetProgrammers = from p in programmers
+	                        where p.IsNewToLINQ
+	                        select new LINQExpert(p);
+	
+Or as a higher-order function via the ``IEnumerable<T>`` interface,
+
+.. code-block:: c#
+
+	var dotNetProgrammers = programmers.Where(p => IsNewToLINQ)
+	                                   .Select(p => new LINQExpert(p));
+									   
+as a way to write expressive, declarative code.
+
+Why is it Useful?
+-----------------
+
+LINQ is an incredibly useful tool for two primary reasons: (1) expressive code, and (2) querying data sources directly from C#/VB via `LINQ Providers <http://stackoverflow.com/a/1568054/1400198>`_  rather than interacting with some obtuse API.
+
+Expressive Code is Valuable
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ask a Ruby programmer why they like Ruby.  Their answer will probably involve something along the lines of "I can express my intent directly in my code".  The same can be said for LINQ.
+
+Imagine you have a list of pets, but want to convert it into a lookup where you can access a pet directly by its ``RFID`` value.
+
+Traditional imperative code:
+
+.. code-block:: c#
+
+	var petLookup = new Dictionary<int, Pet>();
+	
+	foreach (var pet in pets)
+	{
+	    petLookup.Add(pet.RFID, pet);
+	}
+
+Equivalent LINQ expression:
+
+.. code-block:: c#
+
+	var petLookup = pets.ToDictionary(pet => pet.RFIDValue, pet => pet);
```

Comment:
Might this simpler ToDictionary overload work for this scenario?

https://msdn.microsoft.com/en-us/library/vstudio/bb549277(v=vs.100).aspx